docs/branches.md

```diff
@@ -41,6 +40,48 @@ branch `main`. If the local target had also moved compared to `main@origin`
 merged. If one is ahead of the other, then that target will be the new target.
 Otherwise, the local branch will be conflicted (see next section for details).
 
+As of December 2023 Jujutsu tracks and fetches all branches by default, which 
+is confusing users coming from Git and Mercurial. To smoothen the transition 
+branch tracking was introduced. 
+
+### Tracking a branch
+
+To track a branch permanently use `jj branch track <name-of-branch>`. It will 
+now be imported as a local branch until you untrack it or it is deleted on the 
+remote. 
+
+Example:
+
+```sh
+$ # List all available branches, as we want our colleague's branch.
+$ jj git branch list --all
+$ # Find the branch.
+$ jj git branch track <name>
```

Comment:
nit: use `<name>@<remote>` instead?

The `<name>` here must be a remote branch, whereas `<name>` in `jj new <name>` can be either local or remote branch. I think we usually use the local branch name in that case.

Reply:
> nit: use `<name>@<remote>` instead?

Makes sense. I've also added a example. 

> The `<name>` here must be a remote branch, whereas `<name>` in `jj new <name>` can be either local or remote branch. I think we usually use the local branch name in that case.

yes, we prefer the local branch for `jj new` but this is not an issue after tracking the branch.